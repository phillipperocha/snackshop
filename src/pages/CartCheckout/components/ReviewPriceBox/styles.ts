import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    width: 100%;
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.xlarge};
    font-weight: bold;
    height: 8.8rem;
    justify-content: space-between;
    align-items: center;
    padding: 20px 20px 20px 18px;

    span {
      margin-top 8px;
      font-size: ${theme.font.sizes.xxlarge};
    }

    p {
      margin-top: 8px;
    }
  `}
`
