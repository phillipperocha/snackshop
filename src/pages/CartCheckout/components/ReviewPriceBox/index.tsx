import * as S from './styles'

export type ReviewPriceBoxProps = {
  title: string
  price: number
}

const ReviewPriceBox = ({ title, price }: ReviewPriceBoxProps) => (
  <S.Wrapper data-testid="reviewPriceBox">
    <p>{title}</p>
    <span>
      R$
      {price.toLocaleString('pt-BR', {
        minimumFractionDigits: 2,
        maximumFractionDigits: 2,
      })}
    </span>
  </S.Wrapper>
)

export default ReviewPriceBox
