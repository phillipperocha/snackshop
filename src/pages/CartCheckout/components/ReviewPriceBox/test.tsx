import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import ReviewPriceBox from '.'

describe('<ReviewPriceBox />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(
      <ReviewPriceBox title="Total" price={49.89} />
    )

    expect(screen.getByTestId('reviewPriceBox')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
