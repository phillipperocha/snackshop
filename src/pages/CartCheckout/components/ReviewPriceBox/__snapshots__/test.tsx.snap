// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewPriceBox /> should render the heading 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  background: #FF0000;
  color: #FFFFFF;
  font-size: 2.0rem;
  font-weight: bold;
  height: 8.8rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px 20px 20px 18px;
}

.c0 span {
  margin-top: 8px;
  font-size: 3.0rem;
}

.c0 p {
  margin-top: 8px;
}

<div
  class="c0"
  data-testid="reviewPriceBox"
>
  <p>
    Total
  </p>
  <span>
    R$
    49,89
  </span>
</div>
`;
